src/ucp/tag/rndv.c

```diff
@@ -44,6 +44,15 @@ static int ucp_rndv_is_recv_pipeline_needed(ucp_request_t *rndv_req,
     return 1;
 }
 
+static int ucp_rndv_is_put_pipeline_needed(uintptr_t remote_address,
+                                           size_t length, size_t min_get_zcopy,
+                                           size_t max_get_zcopy) {
+    /* fallback to PUT pipeline if remote mem type is non-HOST memory OR
+     * can't do GET ZCOPY */
+    return ((remote_address == 0) || (max_get_zcopy == 0) ||
```

Comment:
check GET_ZCOPY flag in addition to max_get_zopy

Reply:
I think (max_get_zcopy == 0) already capture that. in ep config it is set to 0 by checking GET_ZCOPY flag

Reply:
ok